airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/GlobalMemoryManager.java

```diff
@@ -27,7 +52,13 @@ public long getCurrentMemoryBytes() {
     return currentMemoryBytes.get();
   }
 
+  /**
+   * Requests a block of memory of {@link #BLOCK_SIZE_BYTES}. Return 0 if memory cannot be freed.
+   *
+   * @return the size of the allocated block, in bytes
+   */
   public synchronized long requestMemory() {
+    // todo(davin): what happens if the incoming record is larger than 10MB?
```

Comment:
adding this explicitly to our to do list. this is a case we need to solve this week.